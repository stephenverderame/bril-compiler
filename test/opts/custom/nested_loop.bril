# ARGS: 20
@main(a: int) {
    i: int = const 0;
    run_nest: bool = const false;
.top_head:
    b: bool = lt i a;
    br b .top_body .top_end;
.top_body:
    br run_nest .left .right;
.left:
    j: int = const 0;
.nest_head:
    b2: bool = lt j a;
    br b2 .nest_body .nest_end;
.nest_body:
    one: int = const 1;
    j_inc: int = add j one;
    prod: int = mul i j_inc;
    print prod;
    j: int = add j one;
    jmp .nest_head;
.right:
    zero: int = const 0;
    print zero;
.nest_end:
    run_nest: bool = not run_nest;
    one: int = const 1;
    i: int = add i one;
    jmp .top_head;
.top_end:

}