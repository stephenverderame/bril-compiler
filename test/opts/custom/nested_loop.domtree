    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `/home/sev/code/cs6120-bril/target/debug/bril2cfg --dom`
