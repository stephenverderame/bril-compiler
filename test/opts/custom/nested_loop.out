0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
0
3
6
9
12
15
18
21
24
27
30
33
36
39
42
45
48
51
54
57
60
0
5
10
15
20
25
30
35
40
45
50
55
60
65
70
75
80
85
90
95
100
0
7
14
21
28
35
42
49
56
63
70
77
84
91
98
105
112
119
126
133
140
0
9
18
27
36
45
54
63
72
81
90
99
108
117
126
135
144
153
162
171
180
0
11
22
33
44
55
66
77
88
99
110
121
132
143
154
165
176
187
198
209
220
0
13
26
39
52
65
78
91
104
117
130
143
156
169
182
195
208
221
234
247
260
0
15
30
45
60
75
90
105
120
135
150
165
180
195
210
225
240
255
270
285
300
0
17
34
51
68
85
102
119
136
153
170
187
204
221
238
255
272
289
306
323
340
0
19
38
57
76
95
114
133
152
171
190
209
228
247
266
285
304
323
342
361
380
